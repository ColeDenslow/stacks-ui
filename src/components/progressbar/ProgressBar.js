import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';

import './ProgressBar.css';

function ProgressBar(props) {
  const { size, className, percentCompleted, prefix, suffix, ...extraProps } = props;
  const classNames = cn([
    'stx-progress-bar',
    { 'stx-progress-bar--small': size === 'small' },
    { 'stx-progress-bar--large': size === 'large' },
    className,
  ]);
  const progressStyle = {
    width: `${percentCompleted}%`,
  };

  return (
    <div className={classNames} {...extraProps} role='progressbar' aria-valuenow={percentCompleted} aria-valuemin='0' aria-valuemax='100' >
      <div className='stx-progress-bar__prefix'>
        {prefix}
      </div>
      <div className='stx-progress-bar__container'>
        <div className='stx-progress-bar__fill' style={progressStyle} />
      </div>
      <div className='stx-progress-bar__suffix'>
        {suffix}
      </div>
    </div>
  );
};

ProgressBar.defaultProps = {
  percentCompleted: 0,
};

ProgressBar.propTypes = {
  className: PropTypes.string,
  percentCompleted: PropTypes.number,
  size: PropTypes.oneOf(['small', 'large', '']),
  prefix: PropTypes.node,
  suffix: PropTypes.node,
};

export default ProgressBar;
