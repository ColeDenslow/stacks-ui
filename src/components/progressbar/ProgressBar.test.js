import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import ProgressBar from './ProgressBar';

describe('ProgressBar', () => {
  test('ProgressBar renders with basic props', () => {
    const output = mount(<ProgressBar
      className='the-class'
      percentCompleted={33}
    />);

    expect(toJson(output)).toMatchSnapshot();
  });

  test('ProgressBar renders with render before', () => {
    const output = mount(<ProgressBar
      className='the-class'
      percentCompleted={33}
      prefix={<React.Fragment>
        {'The render before content'}
      </React.Fragment>}
    />);

    expect(toJson(output)).toMatchSnapshot();
  });

  test('ProgressBar renders with render after', () => {
    const output = mount(<ProgressBar
      className='the-class'
      percentCompleted={33}
      suffix={<React.Fragment>
        {'The render before content'}
      </React.Fragment>}
    />);

    expect(toJson(output)).toMatchSnapshot();
  });
});
