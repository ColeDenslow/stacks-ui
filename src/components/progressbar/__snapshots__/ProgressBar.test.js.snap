// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar ProgressBar renders with basic props 1`] = `
<ProgressBar
  className="the-class"
  percentCompleted={33}
>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow={33}
    className="stx-progress-bar the-class"
    role="progressbar"
  >
    <div
      className="stx-progress-bar__prefix"
    />
    <div
      className="stx-progress-bar__container"
    >
      <div
        className="stx-progress-bar__fill"
        style={
          Object {
            "width": "33%",
          }
        }
      />
    </div>
    <div
      className="stx-progress-bar__suffix"
    />
  </div>
</ProgressBar>
`;

exports[`ProgressBar ProgressBar renders with render after 1`] = `
<ProgressBar
  className="the-class"
  percentCompleted={33}
  suffix={
    <React.Fragment>
      The render before content
    </React.Fragment>
  }
>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow={33}
    className="stx-progress-bar the-class"
    role="progressbar"
  >
    <div
      className="stx-progress-bar__prefix"
    />
    <div
      className="stx-progress-bar__container"
    >
      <div
        className="stx-progress-bar__fill"
        style={
          Object {
            "width": "33%",
          }
        }
      />
    </div>
    <div
      className="stx-progress-bar__suffix"
    >
      The render before content
    </div>
  </div>
</ProgressBar>
`;

exports[`ProgressBar ProgressBar renders with render before 1`] = `
<ProgressBar
  className="the-class"
  percentCompleted={33}
  prefix={
    <React.Fragment>
      The render before content
    </React.Fragment>
  }
>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow={33}
    className="stx-progress-bar the-class"
    role="progressbar"
  >
    <div
      className="stx-progress-bar__prefix"
    >
      The render before content
    </div>
    <div
      className="stx-progress-bar__container"
    >
      <div
        className="stx-progress-bar__fill"
        style={
          Object {
            "width": "33%",
          }
        }
      />
    </div>
    <div
      className="stx-progress-bar__suffix"
    />
  </div>
</ProgressBar>
`;
